websites/Z/Zerion/presence.ts

```diff
@@ -0,0 +1,144 @@
+const presence = new Presence({
+	clientId: "1071912828027535462",
+});
+
+async function getStrings() {
+	return presence.getStrings(
+		{
+			viewHome: "general.viewHome",
+			viewing: "general.viewing",
+			search: "general.search",
+			viewGenre: "general.viewGenre",
+			play: "general.watchingVid",
+			pause: "general.paused",
+			searchSomething: "general.searchSomething",
+			buttonViewSeries: "general.buttonViewSeries",
+			viewSeries: "general.viewSeries",
+			watchingSeries: "general.watchingSeries",
+			viewPage: "general.viewPage",
+		},
+		await presence.getSetting<string>("lang").catch(() => "pl")
+	);
+}
+
+enum Assets {
+	Logo = "https://i.imgur.com/MpeS5tp.png",
+	Searching = "https://i.imgur.com/UhPgTRn.png",
+	Viewing = "https://i.imgur.com/RMg2Qgg.png",
+	Play = "https://i.imgur.com/juDzimk.png",
+	Pause = "https://i.imgur.com/GbAMPHI.png",
+}
+
+let video = {
+		duration: 0,
+		currentTime: 0,
+		paused: true,
+	},
+	strings: Awaited<ReturnType<typeof getStrings>>,
+	oldLang: string = null;
+
+function textContent(tags: string) {
+	return document.querySelector(tags)?.textContent?.trim();
+}
+
+presence.on(
+	"iFrameData",
+	(data: { duration: number; currentTime: number; paused: boolean }) => {
+		video = data;
+	}
+);
+
+presence.on("UpdateData", async () => {
+	const presenceData: PresenceData = {
+			largeImageKey: Assets.Logo,
+		},
+		[newLang, privacy, logo, time, buttons] = await Promise.all([
+			presence.getSetting<string>("lang").catch(() => "pl"),
+			presence.getSetting<boolean>("privacy"),
+			presence.getSetting<boolean>("logo"),
+			presence.getSetting<boolean>("time"),
+			presence.getSetting<boolean>("buttons"),
+		]),
+		{ pathname, href } = document.location,
+		path = pathname.split("/"),
+		genres = Array.from(document.querySelectorAll(".genres li[active]"))
+			.map(genre => genre.textContent)
+			.join(", ");
+
+	if (oldLang !== newLang || !strings) {
+		oldLang = newLang;
+		strings = await getStrings();
+	}
+
+	switch (path[1]) {
+		case "":
+			presenceData.details = strings.viewHome;
+			break;
+
+		case "seriale":
+		case "szukaj":
+			presenceData.details = strings.searchSomething;
+			if (genres && !privacy) {
+				presenceData.details = strings.viewGenre;
+				presenceData.state = genres;
+			}
+			presenceData.smallImageKey = Assets.Searching;
+			presenceData.smallImageText = strings.search;
+			break;
+
+		case "serial":
+			presenceData.details = `${strings.viewPage}`;
```

Comment:
```suggestion
			presenceData.details = strings.viewPage
```

Reply:
Forgot to remove. Wanted to insert `${strings.viewPage} ${strings.anime}` but it doesn't look good.